src/profile.js

```diff
@@ -23,45 +25,50 @@ import {withRouter} from 'react-router-dom'
 
     handleUpdate = (event) => {
       event.preventDefault();
-      if(this.completedInput()) {
-        const id = this.props?.location?.state?.id;
-        console.log(id);
-        const myForm = new FormData (document.getElementById("profileForm"));
-        myForm.append("id", id);
-        const myRequest = new Request("http://127.0.0.1:5000/profile", {
-          method: "POST",
-          body: myForm,
-        });
-
-        fetch(myRequest)
-        .then((res) =>
-          res.json())
-        .then((res) => {
-          if(res.response === "Success"){
-            this.props.history.push({
-              pathname: "/questionnaire",
-              state: {id: id}
-            });
-          }
+      if (this.checkAge()) {
+        if (this.completedInput()){
```

Comment:
does it make more sense to check completedInput() before checkAge()? if your age input is not completed does your age calculation still work properly?